src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -117,14 +129,31 @@ def compute_reference(frames: np.ndarray,
             frame[:] = shift_frame(frame=frame, dy=dy, dx=dx)
 
         # Create a new reference image from the highest correlated data.
-        ref_image = max_corr_frames.mean(axis=0).astype(np.int16)
+        with warnings.catch_warnings():
+            # Assuming the motion correction went well, there should be a lot
+            # of empty values in the padded area of the frames. We suppress
+            # warnings for these "Empty Slices" as they are expected.
+            warnings.filterwarnings('ignore', 'Mean of empty slice')
+            ref_image = np.nanmean(max_corr_frames, axis=0)
         # Shift reference image to position of mean shifts to remove any bulk
         # displacement.
         ref_image = shift_frame(
             frame=ref_image,
             dy=int(np.round(-max_corr_ymax.mean())),
             dx=int(np.round(-max_corr_xmax.mean()))
         )
+        # Clip the reference image back down to the original size and remove
+        # any NaNs remaining. Throw warning if a NaN is found.
+        ref_image = ref_image[pad_y:-pad_y, pad_x:-pad_x]
+        if np.any(np.isnan(ref_image)):
+            print(f"Warning: {np.isnan(ref_image).sum()} NaN pixels found in "
+                  "reference image after removing padded values during "
+                  f"iteration {idx}. Likely the image quality is low and "
+                  " shifting frames failed. Setting NaN values to image mean.")
+            ref_image = np.nan_to_num(ref_image,
+                                      nan=np.nanmean(ref_image),
+                                      copy=False)
+        ref_image = ref_image.astype(np.int16)
```

Comment:
Can we cast this to `frames.dtype`?

I know that our data currently comes in as 16 bit integers, but that *could* change someday. One of the things that annoys me about Suite2P is that I'm pretty sure the code expects data to be 16 bit integers but that is not written down anywhere, which is a problem once you run the denoiser and cast the data to floats.

I don't think we are explicitly dependent on the data being 16 bit (except insofar as we are using Suite2P)....